app/src/main/java/timur/gilfanov/messenger/domain/usecase/user/repository/SettingsRepository.kt

```diff
@@ -2,35 +2,46 @@ package timur.gilfanov.messenger.domain.usecase.user.repository
 
 import kotlinx.coroutines.flow.Flow
 import timur.gilfanov.messenger.domain.entity.ResultWithError
+import timur.gilfanov.messenger.domain.entity.user.Identity
 import timur.gilfanov.messenger.domain.entity.user.Settings
 import timur.gilfanov.messenger.domain.entity.user.UiLanguage
-import timur.gilfanov.messenger.domain.entity.user.UserId
 
 /**
- * Repository for managing user settings data.
+ * Repository for managing user settings.
  *
  * Provides access to user settings and operations for updating preferences
- * such as UI language. Coordinates between remote and local data sources
- * to ensure settings are synchronized across devices.
+ * such as UI language.
  */
 interface SettingsRepository {
     /**
      * Observes settings changes for a specific user.
      *
-     * @param userId The unique identifier of the user to observe
+     * @param identity The user identity for which to observe settings
      * @return Flow emitting settings updates or errors
      */
-    fun observeSettings(userId: UserId): Flow<ResultWithError<Settings, UserRepositoryError>>
+    fun observeSettings(
+        identity: Identity,
+    ): Flow<ResultWithError<Settings, GetSettingsRepositoryError>>
 
     /**
      * Changes user's UI language preference.
      *
-     * @param userId The unique identifier of the user
+     * @param identity The user identity for which to change the language
      * @param language The new language preference
      * @return Success or failure with [ChangeLanguageRepositoryError]
      */
     suspend fun changeLanguage(
```

Comment:
Must choose between `UiLanguage` and `Language` for consistency across code base. I prefer first to be specific what language we talking about and it will be easy to introduce language-related code, that not about UI. Open to discussion, if you prefer `Language`. 
```suggestion
    suspend fun changeUiLanguage(
```